assets/src/edit-story/components/panels/textStyle/color.js

```diff
@@ -22,46 +22,48 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { useEffect, useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import { __, _x } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { InputGroup } from '../form';
-import { SimplePanel } from './panel';
-import getCommonValue from './utils/getCommonValue';
+import { InputGroup } from '../../form';
+import BackgroundColorControls from '../shared/backgroundColorControls';
 
-function ColorPanel({ selectedElements, onSetProperties }) {
-  const color = getCommonValue(selectedElements, 'color');
-  const [state, setState] = useState({ color });
-  useEffect(() => {
-    setState({ color });
-  }, [color]);
-  const handleSubmit = (evt) => {
-    onSetProperties(state);
-    evt.preventDefault();
-  };
+function ColorControls({ properties, state, setState }) {
+  const { color, backgroundColor, backgroundOpacity, textOpacity } = properties;
   return (
-    <SimplePanel
-      name="color"
-      title={__('Color', 'web-stories')}
-      onSubmit={handleSubmit}
-    >
+    <>
       <InputGroup
         type="color"
         label={__('Color', 'web-stories')}
         value={state.color}
         isMultiple={color === ''}
         onChange={(value) => setState({ ...state, color: value })}
       />
-    </SimplePanel>
+      <InputGroup
+        type="number"
+        label={__('Text Opacity', 'web-stories')}
+        value={state.textOpacity}
+        isMultiple={'' === textOpacity}
+        onChange={(value) => setState({ ...state, textOpacity: value })}
+        postfix={_x('%', 'Percentage', 'web-stories')}
+        min="1"
+        max="100"
+      />
+      <BackgroundColorControls
```

Comment:
Looks like this is not just "background", so maybe we should rename the component?

Reply:
Currently it is for background only, however, if I understand correctly then these should be replaced by the custom color picker anyway within #271 
I was thinking of creating a separate ColorControls component that could be used for both text and background but not sure if it makes sense with the custom color picker.